src/modules/identity/RadixRemoteIdentity.ts

```diff
@@ -0,0 +1,108 @@
+import RadixECIES from '../crypto/RadixECIES'
+import RadixIdentity from './RadixIdentity'
+import RadixAccount from '../account/RadixAccount'
+import RadixSignature from '../atom/RadixSignature'
+import RadixSerializer from '../serializer/RadixSerializer'
+
+import { RadixAtom, RadixKeyPair } from '../RadixAtomModel'
+
+export default class RadixRemoteIdentity extends RadixIdentity {
+    private TOKEN: string
+    private REMOTE_URL: string
+
+    constructor(readonly keyPair: RadixKeyPair, token, host = 'localhost', port = '54345') {
```

Comment:
What's the flow for creating this supposed to be? When I want to construct it, I don't have the keypair yet - I want to get that through the socket

Reply:
This is the flow:

```js
RadixRemoteIdentity.getRemotePublicKey()
  .then((publicKey) => {
    RadixRemoteIdentity.register('my dApp', 'my dApp description')
      .then((data) => {
         resolve(new RadixRemoteIdentity(RadixKeyPair.fromPublic(publicKey), data.token))
      })
  })
```

The alternative is to open have async code inside the constructor, but that's considered a very bad practice.

Reply:
That's pretty bad from a usability point of view, don't you think? I could never remember how to do that correctly without referring to the documentation.
I agree, async constructor is a bad idea, however what I see most other JS libraries do, including Node itself, is use static constructor methods.

I think one would be appropriate here, something like 

`async RadixRemoteIdentity.createNew('my dApp', 'description'): RadixRemoteIdentity`

You should also add a generateRemoteIdentity() method in the IdentityManager

Reply:
This is much cleaner now:

```js
const identity = await RadixRemoteIdentity.createNew('my dApp', 'my dApp description')
```